EventListener/ContentSecurityPolicyListener.php

```diff
@@ -28,8 +30,10 @@ public function onKernelResponse(FilterResponseEvent $e)
         }
 
         $response = $e->getResponse();
-        $response->headers->add($this->buildHeaders($this->report, true, $this->compatHeaders));
-        $response->headers->add($this->buildHeaders($this->enforce, false, $this->compatHeaders));
+        if (empty($this->hosts) || in_array($e->getRequest()->server->get('HTTP_HOST'), $this->hosts)) {
```

Comment:
Actually the best would be to call `$e->getRequest()->getHost()` here as this supports X-Forwarded-Host and other headers.